src/compiler.rs

```diff
@@ -1283,3 +1310,9 @@ fn generate_entry_point_wrapper<'c>(
     ));
     Ok(())
 }
+
+#[derive(Clone, Debug)]
+enum StatementCompileResult<T> {
+    Processed(T),
+    Deferred,
+}
```

Comment:
I think this is a bit tricky to understand (it took me a while). Could you add some documentation on how the `StatementCompileResult` integrates with `foreach_statement_in_function`?

Reply:
Done.